flow360/component/simulation/models/material.py

```diff
@@ -0,0 +1,67 @@
+from typing import Literal
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+
+
+class MaterialBase(Flow360BaseModel):
+    # Basic properties required to define a material.
+    # For example: young's modulus, viscosity as an expression of temperature, etc.
+    ...
+    name: str = pd.Field()
+
+
+class Gas(MaterialBase):
+    # contains models of getting properites, for example US standard atmosphere model
+    name: Literal["gas"] = pd.Field(frozen=True)
+    viscosity_type: Literal["Sutherland", "constant"] = pd.field("Sutherland")
+    constant_dynamic_viscosity: Optional[ViscosityType] = pd.Field(None)
+    sutherland_constants: Optional[SutherlandConstants] = pd.Field(None)
+    specific_heat_ratio: float = pd.Field()
+    gas_constant: GasConstantType = pd.Field()
+    Prandtl_number: float = pd.Field()
+
+    def viscosity_from_temperature(self, temperature: TemperatureType) -> ViscosityType:
+        """Sutherland's Law"""
+        return constant_dynamic_viscosity
+
+    def speed_of_sound(self, temperature: TemperatureType) -> VelocityType:
+        """Calculates the speed of sound in the air based on the temperature. Returns dimensioned value"""
+        return np.sqrt(self.specific_heat_ratio * self.gas_constant * temperature.to("K")).to("m/s")
+
+
+class SutherlandConstants(Flow360BaseModel):
+    reference_viscosity: ViscosityType = pd.Field()
+    reference_temperature: TemperatureType = pd.Field()
+    effective_temperature: TemperatureType = pd.Field()
+
+
+class Air(Gas):
+    name: Literal["air"] = pd.Field(frozen=True)
+    constant_dynamic_viscosity: ViscosityType = pd.Field(1.825e-5)
+    sutherland_constants: SutherlandConstants = pd.Field(
+        SutherlandConstants(
+            reference_viscosity=1.716e-5, reference_temperature=273, effective_temperature=111
+        )
+    )
+    specific_heat_ratio: float = pd.Field(1.4, frozen=True)
```

Comment:
just FYI that user can still override the value here even frozen.
for example
```python
my_air = Air(specific_heat_ratio = 1000)
```
passes validation. Why do we want these frozen?

Reply:
specific_heat_ratio shouldn't be allowed to change for Air. I need to overwrite with Literal then?